Frameworks/UIKit/UIFontDescriptor.mm

```diff
@@ -29,233 +36,176 @@
+NSString* const UIFontDescriptorFamilyAttribute = static_cast<NSString*>(kCTFontFamilyNameAttribute);
+NSString* const UIFontDescriptorNameAttribute = static_cast<NSString*>(kCTFontNameAttribute);
+NSString* const UIFontDescriptorFaceAttribute = static_cast<NSString*>(kCTFontStyleNameAttribute);
+NSString* const UIFontDescriptorSizeAttribute = static_cast<NSString*>(kCTFontSizeAttribute);
+NSString* const UIFontDescriptorVisibleNameAttribute = static_cast<NSString*>(kCTFontDisplayNameAttribute);
+NSString* const UIFontDescriptorMatrixAttribute = static_cast<NSString*>(kCTFontMatrixAttribute);
+NSString* const UIFontDescriptorCharacterSetAttribute = static_cast<NSString*>(kCTFontCharacterSetAttribute);
+NSString* const UIFontDescriptorCascadeListAttribute = static_cast<NSString*>(kCTFontCascadeListAttribute);
+NSString* const UIFontDescriptorTraitsAttribute = static_cast<NSString*>(kCTFontTraitsAttribute);
+NSString* const UIFontDescriptorFixedAdvanceAttribute = static_cast<NSString*>(kCTFontFixedAdvanceAttribute);
+NSString* const UIFontDescriptorFeatureSettingsAttribute = static_cast<NSString*>(kCTFontFeatureSettingsAttribute);
+NSString* const UIFontDescriptorTextStyleAttribute = @"NSCTFontUIUsageAttribute"; // This matches the reference platform
+
+NSString* const UIFontSymbolicTrait = static_cast<NSString*>(kCTFontSymbolicTrait);
+NSString* const UIFontWeightTrait = static_cast<NSString*>(kCTFontWeightTrait);
+NSString* const UIFontWidthTrait = static_cast<NSString*>(kCTFontWidthTrait);
+NSString* const UIFontSlantTrait = static_cast<NSString*>(kCTFontSlantTrait);
+
+NSString* const UIFontFeatureTypeIdentifierKey = @"CTFeatureTypeIdentifier";
+NSString* const UIFontFeatureSelectorIdentifierKey = @"CTFeatureSelectorIdentifier";
 
 NSString* const SystemFontName = @"Segoe UI";
 
-/**@Status Stub */
-const CGFloat UIFontWeightUltraLight = StubConstant();
-/**@Status Stub */
-const CGFloat UIFontWeightThin = StubConstant();
-/**@Status Stub */
-const CGFloat UIFontWeightLight = StubConstant();
-/**@Status Stub */
-const CGFloat UIFontWeightRegular = StubConstant();
-/**@Status Stub */
-const CGFloat UIFontWeightMedium = StubConstant();
-/**@Status Stub */
-const CGFloat UIFontWeightSemibold = StubConstant();
-/**@Status Stub */
-const CGFloat UIFontWeightBold = StubConstant();
-/**@Status Stub */
-const CGFloat UIFontWeightHeavy = StubConstant();
-/**@Status Stub */
-const CGFloat UIFontWeightBlack = StubConstant();
-
-@implementation UIFontDescriptor {
-@private
-    idretaintype(NSString) _name;
-    float _size;
-    UIFontDescriptorSymbolicTraits _traits;
-}
+const CGFloat UIFontWeightUltraLight = -0.80f;
+const CGFloat UIFontWeightThin = -0.60f;
+const CGFloat UIFontWeightLight = -0.40f;
+const CGFloat UIFontWeightRegular = 0.00f;
+const CGFloat UIFontWeightMedium = 0.23f;
+const CGFloat UIFontWeightSemibold = 0.30f;
+const CGFloat UIFontWeightBold = 0.40f;
+const CGFloat UIFontWeightHeavy = 0.56f;
+const CGFloat UIFontWeightBlack = 0.62f;
```

Comment:
add a comment on these magic numbers about where these came from / how you determined?